### ISSUE TITLE
Overload the * operator of modules.RotationsGate to allow for gate composition

### ISSUE THREAD

**Issue by dominikandreasseitz** [pasqal-io/PyQ#36]
**Update for clarity: This issue is about the Module API.**
We want to be able to do arbitrary gate composition for example for the U gate like:

```python
def U(*args: Any, **kwargs:Any) -> QuantumCircuit:
    return RZ( *args, **kwargs) * RY(*args, **kwargs) * RZ(*args, **kwargs)
```

**Comment by PedroCarreiras** [comment]
I ended up changing the version of python I was using. That allowed me to use `pytest` seamlessly, even though I still haven't manage to get hatch to work. Anyways, I think I got something worth sharing to see if this is the intended functionality. I will open a pull request shortly.

**Comment by nmheim** [comment]
closed via #64
